lib/acts_as_permalink.rb

```diff
@@ -47,37 +47,39 @@ def generate_permalink_for(obj)
           text = text.downcase.strip                  # make the string lowercase and scrub white space on either side
           text = text.gsub(/[^a-z0-9\w]/, "_")        # make any character that is not nupermic or alphabetic into an underscore
           text = text.sub(/_+$/, "").sub(/^_+/, "")   # remove underscores on either end, caused by non-simplified characters
-          text = text[0...obj.class.base_class.instance_variable_get('@permalink_length')]        # trim to length
+          text = text[0...max_length]                 # trim to length
         end
-        
-        # Attempt to find the object by the permalink
-        if obj.class.base_class.send("find_by_#{obj.class.base_class.instance_variable_get('@permalink_column_name')}", text)
-          num = 1
-
-          # If we find the object we know there is a collision, so just add a number to the end until there is no collision
-          while obj.class.base_class.send("find_by_#{obj.class.base_class.instance_variable_get('@permalink_column_name')}", text + num.to_s)
-            num += 1
+
+        # Attempt to find the object by the permalink, and if so there is a collision and we need to de-collision it
+        if obj.class.base_class.where(column_name => text).first
+          candidate_text = nil
+
+          (1..999999).each do |num|
```

Comment:
Technically, if you have a million records with the same name this will start to fail. But it's better than an infinite loop.